### ISSUE TITLE
🐛 [firebase_firestore] MissingPluginException

### ISSUE THREAD

**Issue by haashem** [FirebaseExtended/flutterfire#5083]
## Bug report

**Describe the bug**
I have installed the null safe version of `Firestore` :

```
environment:
  sdk: '>=2.12.0-259.9.beta <3.0.0'

dependencies:
  flutter:
    sdk: flutter
  firebase_core: ^0.8.0-1.0.nullsafety.1
  cloud_firestore: ^0.17.0-1.0.nullsafety.1 
```
but when I try to get a collection document I face `MissingPluginExeption`:
``` 
(MissingPluginException(No implementation found for method Query#get on channel plugins.flutter.io/firebase_firestore))
```

### Steps to reproduce

Install latest null safe version of Firestore as mentioned above, after initializing firebase app, send a simple get request:
```
final _firestore = FirebaseFirestore.instance;
    _firestore
      ..collection('users').get().then((value) {
        print(value);
      }).catchError((e) {
        print(e);
      });
```

### Expected behavior

then completion block should get called without error.

---

## Additional context

Using non null version of Firestore works well
firebase_core: ^0.7.0
cloud_firestore: ^0.16.0+1 

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
[✓] Flutter (Channel beta, 1.26.0-17.6.pre, on macOS 11.0.1 20B29 darwin-x64, locale en-US)
    • Flutter version 1.26.0-17.6.pre at /Users/hashem/Desktop/Xcode_Projects/flutter
    • Framework revision a29104a69b (3 days ago), 2021-02-16 09:26:56 -0800
    • Engine revision 21fa8bb99e
    • Dart version 2.12.0 (build 2.12.0-259.12.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/hashem/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C5020f
    • CocoaPods version 1.10.1

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.19.0

[✓] Connected device (2 available)
    • Android SDK built for x86 (mobile) • emulator-5554                        • android-x86 • Android 7.0
      (API 24) (emulator)
    • iPhone 12 Pro Max (mobile)         • 5925AC79-C5C6-4FEE-B26E-28F6179AB55E • ios         •
      com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.12.0-259.12.beta
Flutter SDK 1.26.0-17.6.pre
firebase_null_safety 1.0.0+1

dependencies:
- cloud_firestore 0.16.0+1 [flutter meta quiver firebase_core firebase_core_platform_interface cloud_firestore_platform_interface cloud_firestore_web]
- cupertino_icons 1.0.2
- firebase_core 0.7.0 [firebase_core_platform_interface flutter quiver meta firebase_core_web]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]

transitive dependencies:
- async 2.5.0 [collection]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.1.0
- charcode 1.2.0
- clock 1.1.0
- cloud_firestore_platform_interface 3.0.2 [flutter meta collection firebase_core plugin_platform_interface]
- cloud_firestore_web 0.3.0+2 [flutter flutter_web_plugins http_parser meta firebase_core firebase_core_web cloud_firestore_platform_interface js]
- collection 1.15.0
- fake_async 1.2.0 [clock collection]
- firebase_core_platform_interface 3.0.1 [flutter meta plugin_platform_interface quiver]
- firebase_core_web 0.2.1+3 [firebase_core_platform_interface flutter flutter_web_plugins meta js]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- http_parser 3.1.4 [charcode collection source_span string_scanner typed_data]
- js 0.6.3
- matcher 0.12.10 [stack_trace]
- meta 1.3.0
- path 1.8.0
- plugin_platform_interface 1.1.0-nullsafety.2 [meta]
- quiver 2.1.5 [matcher meta]
- sky_engine 0.0.99
- source_span 1.8.0 [charcode collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.2.19 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- typed_data 1.3.0 [collection]
- vector_math 2.1.0
```
</details>

---

**Comment by omarvbnet** [comment]
same issue with get_storge plugin

**Comment by markusaksli-nc** [comment]
Reproducible on the latest dev `1.27.0-4.0.pre` with `^0.17.0-1.0.nullsafety.1`.
<details>
 <summary>flutter doctor -v</summary>

```console
[√] Flutter (Channel dev, 1.27.0-4.0.pre, on Microsoft Windows [Version 10.0.19041.804], locale et-EE)
    • Flutter version 1.27.0-4.0.pre at C:\Development\flutter_dev
    • Framework revision f8cd24de95 (6 days ago), 2021-02-16 11:24:17 -0800
    • Engine revision 1d537824d6
    • Dart version 2.13.0 (build 2.13.0-30.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.7042882\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.7042882
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2020.3)
    • IntelliJ at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\203.7148.57
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (4 available)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)           • windows       • windows-x64    • Microsoft Windows [Version 10.0.19041.804]
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 88.0.4324.150
    • Edge (web)                  • edge          • web-javascript • Microsoft Edge 88.0.705.68

• No issues found!
```
</details>
<details>
 <summary>Minimal reproducible code sample</summary>

```dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/material.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  FirebaseApp app = await Firebase.initializeApp();
  assert(app != null);
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Firestore Example'),
      ),
      body: Center(
        child: ElevatedButton(
          onPressed: () => FirebaseFirestore.instance.collection('test').get(),
          child: const Text("Get"),
        ),
      ),
    );
  }
}
```
</details>
<details>
 <summary>Logs</summary>

```console
E/FLTFirestoreMsgCodec(20217): An error occurred while parsing query arguments, this is most likely an error with this SDK.
E/FLTFirestoreMsgCodec(20217): java.lang.NullPointerException: Provided settings must not be null.
E/FLTFirestoreMsgCodec(20217):  at com.google.firebase.firestore.util.Preconditions.checkNotNull(Preconditions.java:147)
E/FLTFirestoreMsgCodec(20217):  at com.google.firebase.firestore.FirebaseFirestore.setFirestoreSettings(FirebaseFirestore.java:193)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestoreMessageCodec.readFirestoreInstance(FlutterFirebaseFirestoreMessageCodec.java:241)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestoreMessageCodec.readValueOfType(FlutterFirebaseFirestoreMessageCodec.java:205)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugin.common.StandardMessageCodec.readValue(StandardMessageCodec.java:317)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugin.common.StandardMessageCodec.readValueOfType(StandardMessageCodec.java:410)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestoreMessageCodec.readValueOfType(FlutterFirebaseFirestoreMessageCodec.java:224)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugin.common.StandardMessageCodec.readValue(StandardMessageCodec.java:317)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestoreMessageCodec.readFirestoreQuery(FlutterFirebaseFirestoreMessageCodec.java:291)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestoreMessageCodec.readValueOfType(FlutterFirebaseFirestoreMessageCodec.java:207)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugin.common.StandardMessageCodec.readValue(StandardMessageCodec.java:317)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugin.common.StandardMessageCodec.readValueOfType(StandardMessageCodec.java:410)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugins.firebase.firestore.FlutterFirebaseFirestoreMessageCodec.readValueOfType(FlutterFirebaseFirestoreMessageCodec.java:224)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugin.common.StandardMessageCodec.readValue(StandardMessageCodec.java:317)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugin.common.StandardMethodCodec.decodeMethodCall(StandardMethodCodec.java:48)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:231)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/FLTFirestoreMsgCodec(20217):  at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:818)
E/FLTFirestoreMsgCodec(20217):  at android.os.MessageQueue.nativePollOnce(Native Method)
E/FLTFirestoreMsgCodec(20217):  at android.os.MessageQueue.next(MessageQueue.java:335)
E/FLTFirestoreMsgCodec(20217):  at android.os.Looper.loop(Looper.java:183)
E/FLTFirestoreMsgCodec(20217):  at android.app.ActivityThread.main(ActivityThread.java:7656)
E/FLTFirestoreMsgCodec(20217):  at java.lang.reflect.Method.invoke(Native Method)
E/FLTFirestoreMsgCodec(20217):  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
E/FLTFirestoreMsgCodec(20217):  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
E/DartMessenger(20217): Uncaught exception in binary message listener
E/DartMessenger(20217): java.lang.IllegalArgumentException: Method call corrupted
E/DartMessenger(20217):         at io.flutter.plugin.common.StandardMethodCodec.decodeMethodCall(StandardMethodCodec.java:52)
E/DartMessenger(20217):         at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:231)
E/DartMessenger(20217):         at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/DartMessenger(20217):         at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:818)
E/DartMessenger(20217):         at android.os.MessageQueue.nativePollOnce(Native Method)
E/DartMessenger(20217):         at android.os.MessageQueue.next(MessageQueue.java:335)
E/DartMessenger(20217):         at android.os.Looper.loop(Looper.java:183)
E/DartMessenger(20217):         at android.app.ActivityThread.main(ActivityThread.java:7656)
E/DartMessenger(20217):         at java.lang.reflect.Method.invoke(Native Method)
E/DartMessenger(20217):         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
E/DartMessenger(20217):         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
E/flutter (20217): [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: MissingPluginException(No implementation found for method Query#get on channel plugins.flutter.io/firebase_firestore)
E/flutter (20217): #0      convertPlatformException (package:cloud_firestore_platform_interface/src/method_channel/utils/exception.dart:13:5)
E/flutter (20217): #1      MethodChannelQuery.get (package:cloud_firestore_platform_interface/src/method_channel/method_channel_query.dart:108:13)
E/flutter (20217): <asynchronous suspension>
E/flutter (20217): #2      Query.get (package:cloud_firestore/src/query.dart:177:9)
E/flutter (20217): <asynchronous suspension>
E/flutter (20217):
```
</details>

**Comment by haashem** [comment]
Fixed in version 1.0.0
